src/roster/views.py

```diff
@@ -1504,5 +1529,6 @@ def get_context_data(self, **kwargs):
         context['current_student_name'] = self.clusive_student.user.first_name if self.clusive_student else "No student"
         context['teacher'] = self.clusive_user
         context['roster'] = self.roster
-        context['data'] = { 'days': 7 }
+        context['panels'] = self.panels
```

Comment:
I didn't do a thorough check, but I do not see `context['panels']` used anywhere in the templates.  Is it used?

I copied it and `self.panels` from the dashboard code for the basic student details template  I have since looked back at the dashboard and they are used there because which panels are shown depends on a number of factors. The dashboard template has a number of a seires of `{% if {{panels.activity %} ... {% elif %} ...` statements for checking which panels are present.

In the case of the student details page, all of the panels are present, so I _think_ it's no longer needed.  I've removed it from the student reactions panel code.

But if it is used in your code, then ir stays.

Reply:
`context['panels']` is not used in my code too. I've removed related lines.

Reply:
👍